File: app/src/main/java/com/platdmit/simplecloudmanager/app/vm/LoadAverageViewModel.java
package com.platdmit.simplecloudmanager.app.vm;

import com.platdmit.simplecloudmanager.domain.models.LoadAverage;
import com.platdmit.simplecloudmanager.domain.repo.ServerLoadAveragesRepo;

import java.util.List;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.LiveDataReactiveStreams;
import androidx.lifecycle.MutableLiveData;
import io.reactivex.rxjava3.processors.BehaviorProcessor;

public class LoadAverageViewModel extends BaseViewModel{
    private static final String TAG = LoadAverageViewModel.class.getSimpleName();
    private ServerLoadAveragesRepo mLoadAveragesRepo;

    private LiveData<List<LoadAverage>> mLoadAverages;
    private MutableLiveData<String> mResultMassage = new MutableLiveData<>();

    private final BehaviorProcessor<List<LoadAverage>> mContentProvider = BehaviorProcessor.create();

    public LoadAverageViewModel(ServerLoadAveragesRepo serverLoadAveragesRepo, long serverId) {
        mLoadAveragesRepo = serverLoadAveragesRepo;

        //Fast fix for prevent overSubscription after resize
        mCompositeDisposable.add(mLoadAveragesRepo.getServerLoadAverages(serverId).subscribe(mContentProvider::onNext));

        mLoadAverages = LiveDataReactiveStreams.fromPublisher(mContentProvider);
    }
    public LiveData<List<LoadAverage>> getLoadAveragesLiveData() {
        return mLoadAverages;
    }

    public MutableLiveData<String> getResultMassage() {
        return mResultMassage;
    }

    @Override
    protected void onCleared() {
        super.onCleared();
        mContentProvider.onComplete();
    }
}

File: app/src/main/java/com/platdmit/simplecloudmanager/vm/LoadAverageViewModel.kt
package com.platdmit.simplecloudmanager.vm

import androidx.hilt.Assisted
import androidx.hilt.lifecycle.ViewModelInject
import androidx.lifecycle.LiveDataReactiveStreams
import androidx.lifecycle.SavedStateHandle
import com.platdmit.domain.repo.ServerLoadAveragesRepo
import com.platdmit.simplecloudmanager.states.LoadAverageState

class LoadAverageViewModel
@ViewModelInject
constructor(
        private val serverLoadAveragesRepo: ServerLoadAveragesRepo,
        @Assisted private val savedStateHandle: SavedStateHandle
) : BaseViewModel<LoadAverageState>() {
    val loadAverageStateLiveData = LiveDataReactiveStreams.fromPublisher(stateProvider)
    val messageLiveData = LiveDataReactiveStreams.fromPublisher(messageProvider)

    init {
        stateProvider.onNext(LoadAverageState.Loading)
    }

    fun setStateIntent(stateIntent: StateIntent){
        when(stateIntent){
            is StateIntent.SetServerId -> {
                setActiveId(stateIntent.id)
            }
            is StateIntent.RefreshResult -> {}
        }
    }

    private fun setActiveId(id: Long){
        //Fast fix for prevent overSubscription after resize
        compositeDisposable.add(
                serverLoadAveragesRepo.getServerLoadAverages(id)
                        .subscribe ({
                            stateProvider.onNext(LoadAverageState.Success(it))
                        },{
                            stateProvider.onNext(LoadAverageState.Error)
                        })
        )
    }

    sealed class StateIntent {
        data class SetServerId(val id: Long) : StateIntent()
        object RefreshResult : StateIntent()
    }
}